src/base/static/js/views/app-view.js

```diff
@@ -208,23 +218,21 @@ module.exports = Backbone.View.extend({
       router: this.options.router,
     }).render();
 
-    this.basemapConfigs = _.find(
-      this.options.leftSidebarConfig.panels,
-      function(panel) {
-        return "basemaps" in panel;
-      },
-    ).basemaps;
-
     // REACT PORT SECTION /////////////////////////////////////////////////////
-    // Instantiate the map.
-    // TODO: Componentize the map module.
-    this.mapView = new MainMap({
-      container: "map",
-      places: this.places,
-      router: this.options.router,
-      mapConfig: mapConfigSelector(store.getState()),
-      store: store,
-    });
+    ReactDOM.render(
+      <Provider store={store}>
+        <MainMap
+          container="map"
+          places={this.places}
+          router={this.options.router}
+          onZoomend={this.onMapZoomEnd.bind(this)}
+          onMovestart={this.onMapMoveStart.bind(this)}
+          onMoveend={this.onMapMoveEnd.bind(this)}
+          onDragend={this.onMapDragEnd.bind(this)}
+        />
+      </Provider>,
+      document.getElementById("map-component"),
```

Comment:
Do we need to have a DOM element to render the map? Strictly speaking there's nothing for React to render in this component-- all UI rendering is handled by the map provider.

Reply:
Shouldn't this element be the same as the `container="map"` prop above? I see that we are passing `container` to our MapProvider to be used as the map's element. And MapboxGL plops the canvas there: https://www.mapbox.com/mapbox-gl-js/api/

It seems like the second argument to ReactDOM.render can be `null` and we can pass in `document.getElementById("map-component")` as the `container` prop. Does that seem reasonable?

Reply:
Ah, I didn't realize that we can pass `null` to `render`-- I think that's a reasonable solution. `map-component` is unnecessary in that case.